### ISSUE TITLE
Basic Constraint not marked critical based on CSR

### ISSUE THREAD

**Issue by cybercorp31** [hashicorp/vault#4529]
Hi

When signing intermediate certificate with vault, it seems that vault changes various aspects present in the CSR and this leads to error in importing the certificate in AWS ACM (PCA).

In order to avoid, we used "use_csr_values=true"

> vault write testpath/root/sign-intermediate csr=@csr.pem ttl=8760h use_csr_values=false

Although this works fine, but the "Basic Constraints" field is not marked as critical (as in the CSR) and this leads to issue in importing certificate in AWS ACM (Private CA).

Ideally "Basic Constraints" should not be dictated by CSR and handling the field is responsibility of the CA. 

There should be a way in which we can configure those fields through configuration or through command line parameters.

**Comment by cybercorp31** [comment]
We are using Vault v0.10.1

If the use_csr_values is not set to true, I have noticed that the subject field provided by the CSR is modified by vault. Thus while importing certificate in AWS ACM (PCA), the validation fails.

For example:

when use_csr_values=true, the subject field looks like following:

`  Subject: C=IN, ST=Mysore, L=Patan Nagar, O=Test, OU=Security, CN=test.internal
`
The above Subject field is ideal one and matches the CSR values.

when use_csr_value is false (default use-case), then subject filed looks like:

`Subject: CN=test.internal
`
Basically the use_csr_values=true does the job and certificate validation is successful at AWS end, but since the "Basic Constraint" is not marked as critical in this case, the import to ACM fails.

If use_csr_values=false, then "Basic Constraint" is set as critical but since CSR values like Subject are ignored, the import fails.

**Comment by jefferai** [comment]
I think this is actually a dupe of #4469 

Any chance you could try with master? I can send a build if you need and if you tell me what platform.

**Comment by catsby** [comment]
Hello - I apologize for the delay in following up here. I'm going to close this issue for now assuming it was addressed in #4469 as stated. Please let us know if you feel otherwise. Thanks!
